File: programmers/c++/네트워크.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

const int MAX = 201;
bool visited[MAX];

void dfs(int start, int n, vector<vector<int>> computers) {
	visited[start] = true;
	for (int i = 0; i < n; i++) {
		if (!visited[i] && computers[start][i])
			dfs(i, n, computers);
	}
	return;
}
int solution(int n, vector<vector<int>> computers) {
	int answer = 0;

	for (int i = 0; i < n; i++) {
		if (!visited[i]) {
			dfs(i, n, computers);
			answer++;
		}
	}

	return answer;
}

File: programmers/python/bfs_dfs/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0]*n
    
    def dfs(node):
        visited[node] = True
        for i in range(n):
            if computers[node][i] and not visited[i]:
                dfs(i)
                
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    return answer
